### DISCUSSION TITLE
If I build and publish to GitHub in a local environment instead of relying on Vercel?

### DISCUSSION THREAD

**Discussion by aiqubits** [avitorio/outstatic#168] [General]
HI, Can I only publish documents using Vercel or a third-party GitHub app? I would like to access Outstatic CMS only in the local development environment or through GitHub pages, publish documents, and synchronously push them to GitHub. Is this feasible?

**Comment by avitorio** [comment]
You can use Outstatic only in the local development environment if you want, but changes are always pushed to GitHub. We suggest Vercel because of the automatic deployments on push. But you can use a different service than Vercel.

**Reply by aiqubits** [reply]
Hi, Is it okay if I don't rely on third-party services and only make local changes and push notifications?

**Reply by avitorio** [reply]
Outstatic curently only works with GitHub.

**Reply by aiqubits** [reply]
I mean is manually push changes to GitHub and then use GitHub pages directly to access the website

**Reply by aiqubits** [reply]
This method should also be possible, but it lacks a mechanism for automatic triggering in the local environment

**Reply by avitorio** [reply]
I think that in this case you have to pull the changes, then build and deploy to GitHub pages.

**Reply by aiqubits** [reply]
Yeah, That is the process, but it requires a service to execute, similar to implementing a lightweight Vercel app that runs locally
